src/Widgets/Banner.vala

```diff
@@ -0,0 +1,167 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2016 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Nathan Dyer <mail@nathandyer.me>
+ */
+
+
+const string BANNER_STYLE_CSS = """
+    .banner {
+        background-color: %s;
+        color: %s;
+    }
+
+    .banner .button {
+        background-color: @base_color;
+    }
+""";
+
+const string DEFAULT_BANNER_COLOR_PRIMARY = "#68758e";
+const string DEFAULT_BANNER_COLOR_PRIMARY_TEXT = "white";
+
+namespace AppCenter.Widgets {
+    public class Banner : Gtk.Button {
+
+        private string _background_color;
+        public string background_color {
+            get {
+                return _background_color;
+            } set {
+                _background_color = value;
+                on_any_color_change ();
+            }
+        }
+        private string _foreground_color;
+        public string foreground_color {
+            get {
+                return _foreground_color;
+            } set {
+                _foreground_color = value;
+                on_any_color_change ();
+            }
+        }
+
+        private Gtk.Box content_box;
+        private Gtk.Label name_label;
+        private Gtk.Label summary_label;
+        private Gtk.Label description_label;
+        private Gtk.Image icon;
+
+        public AppCenterCore.Package? current_package;
+
+        public Banner () {
+            foreground_color = DEFAULT_BANNER_COLOR_PRIMARY_TEXT;
+            background_color = DEFAULT_BANNER_COLOR_PRIMARY;
+            reload_css ();
+            this.height_request = 300;
+
+            // Default AppCenter banner
+            name_label = new Gtk.Label ("Name");
+            name_label.get_style_context ().add_class ("h1");
+            summary_label = new Gtk.Label ("Summary");
+            summary_label.get_style_context ().add_class ("h2");
+            description_label = new Gtk.Label ("Description");
+            description_label.get_style_context ().add_class ("h3");
+            name_label.xalign = 0;
+            summary_label.xalign = 0;
+            description_label.xalign = 0;
+            description_label.margin_top = 25;
```

Comment:
Can we group these instead of having them mixed up? As in everything to do with name_label together, everything to do with summary_label together etc.

We also need to set wrap = true and a max character width (something between 40 and 60)